applicationconfig/client/src/main/java/com/azure/applicationconfig/ConfigurationAsyncClient.java

```diff
@@ -5,37 +5,22 @@
 
 import com.azure.applicationconfig.credentials.ConfigurationClientCredentials;
 import com.azure.applicationconfig.models.ConfigurationSetting;
-import com.azure.applicationconfig.models.SettingFields;
 import com.azure.applicationconfig.models.SettingSelector;
-import com.azure.common.ServiceClient;
-import com.azure.common.exception.ServiceRequestException;
 import com.azure.common.http.HttpPipeline;
-import com.azure.common.http.rest.PagedResponse;
 import com.azure.common.http.rest.Response;
-import com.azure.common.implementation.RestProxy;
-import org.reactivestreams.Publisher;
 import reactor.core.publisher.Flux;
 import reactor.core.publisher.Mono;
 
 import java.net.URL;
-import java.util.Arrays;
-import java.util.Locale;
-import java.util.Objects;
-import java.util.stream.Collectors;
 
 /**
- * Client that contains all the operations for {@link ConfigurationSetting ConfigurationSettings} in Azure Configuration
+ * Asynchronous client that contains all the operations for {@link ConfigurationSetting ConfigurationSettings} in Azure Configuration
  * Store.
  *
  * @see ConfigurationAsyncClientBuilder
  * @see ConfigurationClientCredentials
  */
-public final class ConfigurationAsyncClient extends ServiceClient {
-    private static final String ETAG_ANY = "*";
-
-    private final String serviceEndpoint;
-    private final ConfigurationService service;
-
+public final class ConfigurationAsyncClient extends ConfigurationClientBase<Mono<Response<ConfigurationSetting>>, Flux<ConfigurationSetting>> {
```

Comment:
I think we're lucky in that this client is very simple and only returns two types at this moment. It might be harder to make it extendable though. If the service team decides to bring back `listRevisions` where `Revisions` is another class type, how would we extend this client?

Reply:
Will be changing the implementation method to composition instead of inheritance based on @JonathanGiles's suggestion.